pandas/tests/indexing/multiindex/test_setitem.py

```diff
@@ -196,16 +196,29 @@ def test_multiindex_assignment(self):
         df.loc[4, "d"] = arr
         tm.assert_series_equal(df.loc[4, "d"], Series(arr, index=[8, 10], name="d"))
 
+    def test_multiindex_assignment_single_dtype(self):
+        # GH3777 part 2b
         # single dtype
+        arr = np.array([0.0, 1.0])
+
         df = DataFrame(
             np.random.randint(5, 10, size=9).reshape(3, 3),
             columns=list("abc"),
             index=[[4, 4, 8], [8, 10, 12]],
+            dtype=np.int64,
         )
 
+        # arr can be losslessly cast to int, so this setitem is inplace
         df.loc[4, "c"] = arr
-        exp = Series(arr, index=[8, 10], name="c", dtype="float64")
-        tm.assert_series_equal(df.loc[4, "c"], exp)
+        exp = Series(arr, index=[8, 10], name="c", dtype="int64")
+        result = df.loc[4, "c"]
+        tm.assert_series_equal(result, exp)
```

Comment:
Should we use a view here to show, that the underlying object changed?

Reply:
good call, turns out the ArrayManager subs in a new object

Reply:
updated+green